### PR TITLE
[Feature] Support pooling models on 310P platform

### PR DESCRIPTION
**Pull request by Jeaniowang**
### What this PR does / why we need it?
this PR adds support for pooling models (e.g., embedding, classification, scoring) on the Ascend 310P platform. It implements the necessary attention mask generation and flash attention forward pass for non-causal attention used by these models

### Does this PR introduce _any_ user-facing change?
Yes, pooling models are now supported on 310P.

### How was this patch tested?
Added E2E tests for classification, embedding, and scoring models on 310P.

resolve thess issues: [6471](https://github.com/vllm-project/vllm-ascend/issues/6471) [7363](https://github.com/vllm-project/vllm-ascend/pull/7363) [6199](https://github.com/vllm-project/vllm-ascend/issues/6199)

- vLLM version: v0.20.2
- vLLM main: https://github.com/vllm-project/vllm/commit/1ac10f159a09897baada01b14b6a0dd6442aefd6

### CODE REVIEW

**Comment by Jeaniowang** [comment]
rebuild

**Comment by Jeaniowang** [comment]
rebuild

File: vllm_ascend/attention/attention_mask.py [hunk 1]
```diff
@@ -72,7 +72,16 @@ def get_pcp_mla_mask(self, dtype: torch.dtype):
...
+            if sliding_window is not None:
+                mask = torch.ones(2048, 2048, dtype=torch.bool)
+                triu_mask = torch.triu(mask, diagonal=1).to(self.device)
+                tril_mask = torch.tril(mask, -sliding_window).to(self.device)
+                self.swa_mask = triu_mask + tril_mask
+        return self.swa_mask
+
+    def get_attention_mask(self, causal: bool, model_config: ModelConfig):
         if model_config.runner_type == "pooling":
             return self.get_attn_mask(2048, torch.bool)
```
[the remark this answers is not in the document]
Comment (Jeaniowang) [reply vllm_ascend/attention/attention_mask.py:77]: BERT-based embeddings no need config attention-mask

**Review by Tflowers-0129 — APPROVED** [review]

**Review by realliujiaxu — APPROVED** [review]
